2020/8xxx/CVE-2020-8908.json

```diff
@@ -43,7 +43,7 @@
         "description_data": [
             {
                 "lang": "eng",
-		"value": "A temp directory creation vulnerability exist in Guava versions prior to 30.0 allowing an attacker with access to the machine to potentially access data in a temporary directory created by the Guava com.google.common.io.Files.createTempDir(). The permissions granted to the directory created default to the standard unix-like /tmp ones, leaving the files open. We recommend updating Guava to version 30.0 or later, or update to Java 7 or later, or to explicitly change the permissions after the creation of the directory if neither are possible."
+		"value": "A temp directory creation vulnerability exists in all versions of Guava, allowing an attacker with access to the machine to potentially access data in a temporary directory created by the Guava API com.google.common.io.Files.createTempDir(). By default, the created directory is world-readable (readable by an attacker with access to the system). The method in question has been marked @Deprecated in versions 30.0 and later and should not be used. For Android developers, we recommend choosing a temporary directory API provided by Android, such as context.getCacheDir(). For other Java developers, we recommend migrating to the Java 7 API java.nio.file.Files.createTempDirectory() which explicitly configures permissions of 700, configuring the Java runtime's java.io.tmpdir system property to point to a location whose permissions are appropriately configured."
```

Comment:
The only change I would recommend is offering a workaround option of setting the system property `java.io.tmpdir` to a "safe" directory.

Reply:
I might also change this: 

> By default, the created directory is world-readable (readable by an attacker with access to the system).

To be something like:

> By default, on unix-like systems (and very old versions of MacOS), the created directory is world-readable (readable by an attacker with access to the system)

Reply:
The "very old versions of MacOS" came from this other vulnerability I found in Apache Groovy.
I didn't write the "very old versions of MacOS" line, I'm guessing someone in the Apache foundation did that research and put that in there.

So, I can't back up that fact, but I could send an email if someone feels that that's important to verify.

http://groovy-lang.org/security.html#CVE-2020-17521

Reply:
> The only change I would recommend is offering a workaround option of setting the system property `java.io.tmpdir` to a "safe" directory.

It already mentions that, I think `[[or]] configuring the Java runtime's java.io.tmpdir system property to point to a location whose permissions are appropriately configured` - it's just missing an or.